src/Type.cpp

```diff
@@ -690,26 +736,34 @@ Type::make_one_union_field(vector<const Type*> &fields, vector<CVQualifiers> &qf
 	}
 	else {
 		size_t i;
-		vector<Type*> ok_types;
-		// filter out struct types containing bit-fields. Their layout is implementation
-		// defined, we don't want to mess with them in unions for now
-		for (i=0; i<AllTypes.size(); i++) {
-			if (!AllTypes[i]->has_bitfields()) {
-				ok_types.push_back(AllTypes[i]);
-			}
-		}
-
-		// find of struct types
-		vector<Type*> struct_types;
-		for (size_t i=0; i<ok_types.size(); i++) {
-			if (ok_types[i]->eType == eStruct) {
-				struct_types.push_back(ok_types[i]);
+			vector<Type*> ok_nonstruct_types;
+			vector<Type*> struct_types;
+			// vector<Type*> union_types;
+			for (i = 0; i < AllTypes.size(); i++) {
+                if ((AllTypes[i]->eType == eStruct) || (AllTypes[i]->eType == eUnion)){
+					// filter out structs and unions containing bit-fields. Their layout is implementation 
+					// defined, we don't want to mess with them in unions for now
+					if (!AllTypes[i]->has_bitfields()) {
+						// filter out structs/unions with assign operators (C++ only) 
+                        if (!(AllTypes[i]->has_implicit_nontrivial_assign_ops())) {
+							if (AllTypes[i]->eType == eStruct){
+								struct_types.push_back(AllTypes[i]);
+							}
+							else{ // union
+								// no union in union currently
+								// union_types.push_back(AllTypes[i]);
+							}
+						}
+					}
+				}
+				else{
+					ok_nonstruct_types.push_back(AllTypes[i]);
+				}
```

Comment:
Can we reduce the level of nested statements? E.g.,

    if (AllTypes[i]->eType != eStruct && AllTypes[i]->eType != eUnion) {
        ok_nonstruct_types.push_back(AllTypes[i]);
        continue;
    }
    if (!AllTypes[i]->has_bitfields())
        continue;
    if (!AllTypes[i]->has_implicit_nontrivial_assign_ops())
        continue;
    ...